content/http-handler-abi.md

```diff
@@ -808,7 +810,8 @@ below, and the `len` of the URI would be returned:
 ;; set_uri overwrites the URI with one read from memory, e.g.
 ;; "/v1.0/hi?name=panda".
 ;;
-;; Note: The URI may include query parameters.
+;; Note: The URI may include query parameters. The host MUST ensure URI passed
```

Comment:
is passed the right word?  below we say "A host who fails to set the URI will trap"...

if we are saying the guest must pass this encoded, we can clarify note below that "A host who receives an unencoded URI or otherwise fails to set the URI will trap"

Otherwise, maybe this line should be something like.. "Note: ... The guest MAY pass the URI unencoded. When a host receives an unencoded URI, it MUST encode it."

Reply:
Interesting. How would the host know the URI is unencoded in either case? I think this takes us to the point where we require the guest to encode the URI and we just trust it is encoded because I don't see a way to determine whether `?name=chip&dale` is encoded or not as the spec does not define how to deal with this syntax see https://www.rfc-editor.org/rfc/rfc3986#page-23

Reply:
so I think we end up needing to have the guest encode it. so maybe something like my former comment. though if we have no way to tell if it is encoded or not, then the host will have no way to strictly enforce it.

 "the guest MUST encode the entire URI (e.g. path in XX encoding and query in YY encoding)" then later "A host who receives an unencoded URI or otherwise fails to set the URI MAY trap"  the latter thing MAY as if it is true we can't tell if something is encoded then we can't say the host must trap on bad encoding.

Reply:
I'm good with the current text or adding some clarification of possible trap, either's good for me. I guess there are probably other cases that using the ABI in an undefined way can cause panics (e.g. writing a non-utf8 string) so I wonder if we want to explicitly write out all of them or to keep it as a general principle. Though URL encoding may be normal enough (as opposed to non-utf8 string) that it's worth documenting anyways